pkg/diagnostics/workflow_metrics.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package diagnostics
+
+import (
+	"context"
+
+	diagUtils "github.com/dapr/dapr/pkg/diagnostics/utils"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+var (
+	elapsedKey       = tag.MustNewKey("elapsed")
+	reminderTypeKey  = tag.MustNewKey("reminder_type")
+	executionTypeKey = tag.MustNewKey("execution_type")
+)
+
+const (
+	StatusSuccess   = "success"
+	StatusFailed    = "failed"
+	StatusRetryable = "retryable"
+	CreateWorkflow  = "create_workflow"
+	GetWorkflow     = "get_workflow"
+	AddEvent        = "add_event"
+	PurgeWorkflow   = "purge_workflow"
+	Activity        = "activity"
+	Workflow        = "workflow"
+
+	WorkflowEvent = "event"
+	Timer         = "timer"
+
+	ComponentName = "dapr"
+)
+
+type workflowMetrics struct {
+	// Total Successful and Failed workflow operation requests
```

Comment:
What all constitutes workflow operation?
Create/ get/ Delete etc. on workflow? If yes, can we please add to the comment here?

Reply:
Workflow Operations Requests consist of:
- Create workflow
- Get workflow
- Purge workflow
- Add Events
I have added the description of them, should I provide full description for every metric ?